contrib/db_pools/lib/src/config.rs

```diff
@@ -0,0 +1,48 @@
+use rocket::serde::{Deserialize, Serialize};
+
+use crate::Error;
+
+/// A base `Config` for any `Pool` type.
+///
+/// For the following configuration:
+///
+/// ```toml
+/// [global.databases.my_database]
+/// url = "postgres://root:root@localhost/my_database"
+/// pool_size = 10
+/// ```
+///
+/// ...the following struct would be passed to [`Pool::initialize()`]:
+///
+/// ```rust
+/// # use rocket_db_pools::Config;
+/// Config {
+///     url: "postgres://root:root@localhost/my_database".into(),
+///     pool_size: Some(10),
+/// };
+/// ```
+///
+/// If you want to implement your own custom database adapter and need some more
+/// configuration options, you may need to define a custom `Config` struct.
+///
+/// [`Pool::initialize()`]: crate::Pool::initialize
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+#[serde(crate = "rocket::serde")]
+pub struct Config {
```

Comment:
We shouldn't need this structure at all. Each implementation should define their own config structure.

See my other comments in config.